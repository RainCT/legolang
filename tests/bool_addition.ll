bool a = false
print a             // false

int b = a + a
print b             // 0

a = true
b = a + a
print b             // 2

b = 2 + a
print 3             // 3

b = 3
print b + a         // 4

/* Check some int to bool casting while we are at it */
print ""

bool b0 = b + a
print b0            // true

int z = 0
bool b1 = z
print b1            // false
