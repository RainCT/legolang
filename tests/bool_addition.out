false
0
2
3
4

true
false
